scripts/check_doc_ownership.py

```
@@ -531,13 +698,77 @@ def main():
+            if not older or older == newer:
+                continue
+            for _line, doc, old, new in reassigned_docs(older, newer):
+                # FIRST owner wins. A doc captured twice on one branch
+                # (A -> B -> C) would otherwise be reported as having left B,
+                # which is the thief from the first move; the item that
+                # actually lost its documentation is A, and a reader who
+                # repairs the item the message names leaves A bare.
+                candidates.setdefault(doc, (old, new))
+        if not candidates:
+            continue
+        losses_keys = [(l[0], l[1]) for l in losses]
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)